### PR TITLE
Migrate ca-vote, ready for review.

### PR DESCRIPTION
**Pull request by tapina**
.Fairly simple web component

### CODE REVIEW

File: components/ca-vote.css [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+ca-vote button::before {
+    font-family: fontawesome;
+    position: relative;
+    left: -10px;
+    font-size: 20px;
+}
+
+ca-vote[data-status="enabled"] button{
+    background: #4CAF50;
```
Comment (nealeu) [inline components/ca-vote.css]: Missing whitespace before { 
Just sayin'...

File: components/ca-vote.js [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+            return ['id'];
+        }
+
+        /**
+         * @description Executes when any attribute is changed on the element
+         * @type {Event}
+         * @param {string} attrName - the name of the attribute to have changed
+         * @param {string} oldVal - the old value of the attribute
+         * @param {string} newVal - the new value of the attribute
+         * @returns {void} void
```
Comment (nealeu) [inline components/ca-vote.js]: Inconsistent.  Is it void or undefined?

File: components/ca-vote.js [hunk 3]
```diff
@@ -0,0 +1,156 @@
+/* global api */
+define(['./helpers/create-element.js', './helpers/clear-element.js', 'document-register-element'], (createElement, clearElement) => {
+    const componentName = 'ca-vote';
```
Comment (kevmarchant) [inline components/ca-vote.js]: no need to put the component name at the top, only to use it once at the bottom, think we're just refactoring that one out

File: components/ca-vote.js (hunk 2 above)
Comment (kevmarchant) [inline components/ca-vote.js]: I prefer {void} void

File: components/ca-vote.js [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+
+            if (localStorage[this.storageKey]) {
+                this.status = 'disabled';
+                this.label = 'Thanks for your vote';
+            } else {
+                this.status = 'enabled';
+                this.label = 'Vote for this feature';
+            }
+
+            this.button = createElement(this, 'button', null, this.label);
```
Comment (kevmarchant) [inline components/ca-vote.js]: I think this obfuscation is unnecessary, but happy to have that discussion come refactoring time.

File: components/ca-vote.js (hunk 2 above)
Comment (tapina) [inline components/ca-vote.js]: I suppose technically it does return undefined. Although http://stackoverflow.com/questions/4759175/how-to-return-void-in-jsdoc.

File: components/ca-vote.js (hunk 3 above)
Comment (tapina) [inline components/ca-vote.js]: It's used in a few places, hence the factoring out into a constant.

File: components/ca-vote.js (hunk 4 above)
Comment (tapina) [inline components/ca-vote.js]: Agree for this one, the helper is just adding bloat.

**Comment by tapina** [comment]
@nealeu are you happy with this now?

**Pull request merged**